### PR TITLE
onevent optimization

### PR DESCRIPTION
**Pull request by harshit-bs**
Thanks in advance for your contribution. Please follow the below steps in submitting a pull request, as it will help us with reviewing it quicker.

- [ ] Create a new branch from master (e.g. `features/my-new-feature` or `issue/123-my-bugfix`);
- [ ] If you're fixing a bug also create an issue if one doesn't exist yet;
- [ ] If it's a new feature explain why do you think it's necessary. Please check with the maintainers beforehand to make sure it is something that we will accept. Usually we only accept new features if we feel that they will benefit the entire community;
- [ ] Please avoid sending PRs which contain drastic or low level changes. If you are certain that the changes are needed, please discuss them beforehand and indicate what the impact will be;
- [ ] If your change is based on existing functionality please consider refactoring first. Pull requests that duplicate code will most likely be ignored;
- [ ] Do not include changes that are not related to the issue at hand;
- [ ] Follow the same coding style with regards to spaces, semicolons, variable naming etc.;
- [ ] Always add unit tests - PRs without tests are most of the times ignored.

### CODE REVIEW

File: lib/runner/cli/cli.js [hunk 1]
```diff
@@ -294,8 +277,23 @@ class CliRunner {
...
     if (this.isOnEventCallbackExistsInGlobal()) {
-      NightwatchEventHub.setOnEventFunction(this.customisedOnEvent.bind(this));
       NightwatchEventHub.runner = this.testRunner.type;
+      NightwatchEventHub.isAvailable = true;
+      
+      const {globals} = this.test_settings;
+      const {plugins} = this.globals;
+  
+      if (Utils.isFunction(globals.onEvent)) {
+        globals.onEvent(NightwatchEventHub);
```
Comment (swrdfish) [inline lib/runner/cli/cli.js:287]: we should use the name `setEventHandlers` instead of `onEvent`

File: test/extra/cucumber-config-events.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  test_runner: {
+    type: 'cucumber',
+    options: {
+      feature_path: path.join(__dirname, '../cucumber-integration-tests/sample_cucumber_tests/withexpect/sample.feature')
+    }
+  },
+  output: false,
+  silent: false
+};
```
Comment (harshit-bs) [inline test/extra/cucumber-config-events.js:57]:
```suggestion
};

```

File: test/cucumber-integration-tests/testCucumberTestsWithExpect.js [hunk 3]
```diff
@@ -46,4 +46,32 @@ describe('Cucumber integration with .expect APIs', function() {
...
+      output: false,
+      globals: {
+        waitForConditionTimeout: 10,
+        waitForConditionPollInterval: 50
+      }
+    })).then(failures => {
+      assert.strictEqual(failures, true, 'Cucumber should have test failures');
+    });
+  });
 });
```
Comment (harshit-bs) [inline test/cucumber-integration-tests/testCucumberTestsWithExpect.js:77]:
```suggestion
});

```

File: test/src/runner/testEventReporter.js [hunk 4]
```diff
@@ -105,7 +97,7 @@ describe('testNightwatchEventReporter', function() {
     } catch (error) {
       possibleError = error;
     }
-    
+
```
Comment (harshit-bs) [inline test/src/runner/testEventReporter.js:100]:
```suggestion
    
```

**Pull request merged**